independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java

```diff
@@ -141,22 +142,28 @@ public BeanFactory.BeanInstance<?> apply(Class<?> aClass) {
             }
             Map<String, RequestMapper<RuntimeResource>> mappersByMethod = new RuntimeMappingDeployment(templates)
                     .buildClassMapper();
-            mappersByMethod.forEach((method, mapper) -> {
-                for (RequestMapper.RequestPath<RuntimeResource> path : mapper.getTemplates()) {
-                    if ((clazz.getIsDisabled() != null) && clazz.getIsDisabled().get()) {
-                        String templateWithoutSlash = path.template.template.startsWith("/")
-                                ? path.template.template.substring(1)
-                                : path.template.template;
-                        String fullPath = clazz.getPath().endsWith("/") ? finalPrefix + clazz.getPath() + templateWithoutSlash
-                                : finalPrefix + clazz.getPath() + "/" + templateWithoutSlash;
-                        if (!disabledEndpoints.containsKey(fullPath)) {
-                            disabledEndpoints.put(fullPath, new ArrayList<>());
+            boolean isResourceClassDisabled = (clazz.getIsDisabled() != null) && clazz.getIsDisabled().get();
+            mappersByMethod.forEach(new BiConsumer<>() {
+                @Override
+                public void accept(String method, RequestMapper<RuntimeResource> mapper) {
+                    for (int i = 0; i < mapper.getTemplates().size(); i++) {
+                        RequestMapper.RequestPath<RuntimeResource> path = mapper.getTemplates().get(i);
+                        if (isResourceClassDisabled) {
```

Comment:
If `isResourceClassDisabled` is `true`, then this `forEach` iterates over `mappersByMethod`, then over their `mapper.getTemplates` and then does nothing? Maybe that if should be before `forEach`?

Reply:
Makes perfect sense!